com.unity.netcode.gameobjects/Runtime/Utility/Ref.cs

```diff
@@ -0,0 +1,36 @@
+using System.Runtime.CompilerServices;
+
+namespace Unity.Netcode
+{
+    public struct Ref<T> where T : unmanaged
```

Comment:
as far as I can tell, this is just a wrapper for raw pointers, right?
can we just use raw pointers in structs (reader/writer/serializer etc.) instead of this new wrapper type since this won't give us any extra guarantee or safety benefits anyway?

Reply:
This isn't new... this is from the FastBufferReader branch. We already had this discussion there, but reiterating my stance there, this is just getting rid of some boilerplate on having to put things in fixed() blocks, and IMO, is easier to comprehend for people reading our code who have less experience with pointers. It doesn't provide any functionality or safety, it just reduces boilerplate and improves readability... IMO.

Reply:
> This isn't new... this is from the FastBufferReader branch. We already had this discussion there

oh, I didn't want to resurrect but I accidentally did :P

> is easier to comprehend for people reading our code who have less experience with pointers

a dev not knowing what a pointer is wouldn't use this util properly without understanding details anyway.

> It doesn't provide any functionality or safety, it just reduces boilerplate and improves readability...

fair but might not need to ship out there as a construct but sure :)

Reply:
Does it make you feel less squeamish if I make this internal? As I mentioned elsewhere, I am bad about making things public that should be internal. This should be internal.

Reply:
exposing it to users publicly would be a serious concern — using it internally for us is a slight concern of mine. I would still prefer/suggest sticking with unsafe block so that we can be more delicate, careful and intentional about these kind of things.
(since it'll be internal) we can bring this onto table again later and we can discuss whether or not to adapt this pattern everywhere in the codebase. one thing I want to avoid is to use `Ref<T>` _sometimes_ and not **always**. I'd prefer sticking with barebones but that's a personal stance. again, let's keep it and have that conversation later.

Reply:
It's now internal.